rules-reviewed/eap7/eap6/hsearch.windup.xml

```diff
@@ -0,0 +1,880 @@
+        </rule>
+        <rule id="hsearch-00009">
+            <when>
+                <javaclass references="org.hibernate.search.spi.SearchFactoryBuilder" />
+            </when>
+            <perform>
+                <hint title="Hibernate Search 5 package change for SearchFactoryBuilder" effort="1" severity="mandatory">
+                    <message>The class `org.hibernate.search.spi.SearchFactoryBuilder` were replaced by `org.hibernate.search.spi.SearchIntegratorBuilder` class.</message>
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_hibernate_search_changes"
+                        title="Changes in Hibernate Search 5.x" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hsearch-00010">
+            <when>
+                <javaclass references="org.hibernate.search.spi.SearchFactoryIntegrator" />
+            </when>
+            <perform>
+                <hint title="Hibernate Search 5 package change for SearchFactoryIntegrator" effort="1" severity="mandatory">
+                    <message>The class `org.hibernate.search.spi.SearchFactoryIntegrator` were replaced by `org.hibernate.search.spi.SearchIntegrator` class.</message>
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_hibernate_search_changes"
+                        title="Changes in Hibernate Search 5.x" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hsearch-00011">
+            <when>
+                <javaclass references="org.hibernate.search.Version" />
+            </when>
+            <perform>
+                <hint title="Hibernate Search 5 package change for Version" effort="1" severity="mandatory">
+                    <message>The class `org.hibernate.search.Version` were replaced by `org.hibernate.search.engine.Version` class.</message>
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_hibernate_search_changes"
+                        title="Changes in Hibernate Search 5.x" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+        </rule>
+        <!-- Deprecated things -->
+        <rule id="hsearch-00100">
+            <when>
+                <javaclass references="org.hibernate.search.cfg.NumericFieldMapping(java.lang.String, org.hibernate.search.cfg.PropertyDescriptor, org.hibernate.search.cfg.EntityDescriptor, org.hibernate.search.cfg.SearchMapping)">
+                    <location>CONSTRUCTOR_CALL</location>
```

Comment:
my bad, it should be read as use `NumericFieldMapping(String, PropertyDescriptor, EntityDescriptor, SearchMapping)` instead of `NumericFieldMapping(PropertyDescriptor, EntityDescriptor, SearchMapping)`